sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java

```diff
@@ -645,6 +642,46 @@ private void stopConnectionMonitoring() {
         return handle;
     }
 
+    public SlackMessageHandle<SlackMessageReply> sendCodeSnippetToUser(SlackUser user, String title, String snippet, String language) {
+        SlackChannel iMChannel = getIMChannelForUser(user);
+        return sendCodeSnippet(iMChannel, title, snippet, language);
+    }
+
+    public SlackMessageHandle<SlackMessageReply> sendCodeSnippet(SlackChannel channel, String title, String snippet, String language) {
+        SlackMessageHandle<SlackMessageReply> handle = new SlackMessageHandle<>(getNextMessageId());
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("token", authToken);
+        arguments.put("channels", channel.getId());
+        arguments.put("title", title);
+        arguments.put("filetype", language);
+        postSlackCommandWithFile(arguments, snippet.getBytes(), title, FILE_UPLOAD_COMMAND, handle);
```

Comment:
using title as upload file name is ok ? (e.g. any restrictions in characters usable ?)

Reply:
@Geanik what do you think about file name question? we can generate GUID-like file name if @gluck think it's better like that?